DeepMRSeg/deepmrseg_test.py

```diff
@@ -432,140 +461,387 @@ def _main():
 	#ENDIF
 
+	print("Temp folder set to : " + tmpDir)
+	_sys.stdout.flush()
+	
+
 	### Print parsed args
-	print( "\nFile List \t: %s" % (FLAGS.sList) )
-	print( "ID Column \t: %s" % (FLAGS.idCol) )
-	print( "Output Column \t: %s" % (FLAGS.outCol) )
-	print( "Ref Modality \t: %s" % (FLAGS.refMod) )
-	print( "Other Mods \t: %s" % (otherMods) )
-	print( "Num of Mods \t: %d" % (num_modalities) )
+	if IOType == 1:
+		print( "\nInput image(s) \t: %s" % (' , '.join(FLAGS.inImg)) )
+		print( "\nOutput image \t: %s" % (FLAGS.outImg) )
+	if IOType == 2:
+		print( "\nInput dir \t: %s" % (FLAGS.inDir) )
+		print( "\nOutput dir \t: %s" % (FLAGS.outDir) )
+		print( "\nOutput suffix \t: %s" % (FLAGS.outSuff) )
+	if IOType == 3:
+		print( "\nFile List \t: %s" % (FLAGS.sList) )
+
 	print( "Data Prepped \t: %d" % (FLAGS.prep) )
 	
 	print("\nModel Dir(s) \t: %s" % (FLAGS.mdlDir))
-	print( "ROI csv \t: %s" % (FLAGS.roi) )
 
-	print("\nNum of Classes \t: %d" % (FLAGS.num_classes))
-	print("Rescale Method \t: %s" % (FLAGS.rescale))
-	print("XY width \t: %d" % (FLAGS.xy_width))
-	print("Voxel Size \t: %f" % (FLAGS.ressize))
-	print("Orientation \t: %s" % (FLAGS.reorient))
 	print("Batch Size \t: %s" % (FLAGS.batch))
 	
 	print("\nOutput probs \t: %d" % (FLAGS.probs))
 	
-	### Create temp dir, if needed
 	import csv as _csv
 	from concurrent.futures import ThreadPoolExecutor as _TPE
 
+	###############################################################
+	#### For direct I/O options: Create list file with in/out images 
+	if IOType == 1:
+
+		# Generate list of columns
+		numMod = len(FLAGS.inImg)   
+		modCols = 'Mod_'
+		modCols = [modCols + str(i+1) for i in range(numMod)]
+		columns = _np.array(['ID'] + modCols + ['OutImg']).reshape(1,-11)
+
+		# Add absolute path to images
+		FLAGS.inImg = list(map(_os.path.abspath, FLAGS.inImg))   
+		FLAGS.outImg = _os.path.abspath(FLAGS.outImg)
+
+		# Create csv data
+		data=_np.array(['Scan1'] + FLAGS.inImg + [FLAGS.outImg]).reshape(-1, numMod+2)
+
+		# Write csv file
+		outCsv = _os.path.join(tmpDir, 'deepmrseg_test_filelist.csv')
+		with open(outCsv, 'w') as csvfile:
+			csvw = _csv.writer(csvfile)
+			csvw.writerows(columns)
+			csvw.writerows(data)
+
+		# Set flag for image list
+		FLAGS.sList = outCsv
+
+	if IOType == 2:
+
+		# Generate list of columns
+		numMod = 1          ## Multi-modal data is not implemented for "input from folder"
+		modCols = 'Mod_'
+		modCols = [modCols + str(i+1) for i in range(numMod)]
+		columns = _np.array(['ID'] + modCols + ['OutImg']).reshape(1,-1)
+
+		# Add absolute path to in/out dirs
+		FLAGS.inDir = _os.path.abspath(FLAGS.inDir)
+		FLAGS.outDir = _os.path.abspath(FLAGS.outDir)
+		
+		# Add image extension to out suffix (.nii.gz for now)
+		if FLAGS.outSuff.endswith('.nii.gz') == False:
+			FLAGS.outSuff = FLAGS.outSuff + '.nii.gz'
+			
+		# Create data frame
+		#   Get list of all .nii.gz images in the input dir
+		data = []
+		subInd = 1
+		for (dir, _, files) in _os.walk(FLAGS.inDir):
```

Comment:
Instead of assuming all files in inDir are input files, why not accept a suffix to look for? For e.g., a folder may have both T1 and FL, but we may only need T1 scans as input.